builds/checkin/rust-test-modules.yaml

```diff
@@ -50,6 +52,10 @@ jobs:
           failTaskOnFailedTests: true
         condition: succeededOrFailed()
       - bash: scripts/linux/generic-rust/format.sh --project-root "test/modules/generic-mqtt-tester"
-        displayName: Format Code
+        displayName: Format Code (Generic MQTT Tester)
       - bash: scripts/linux/generic-rust/clippy.sh --project-root "test/modules/generic-mqtt-tester"
-        displayName: Clippy
+        displayName: Clippy (Generic MQTT Tester)
+      - bash: scripts/linux/generic-rust/format.sh --project-root "test/modules/obsagent-client"
+        displayName: Format Code (Observability-Agent Client)
+      - bash: scripts/linux/generic-rust/clippy.sh --project-root "test/modules/obsagent-client"
+        displayName: Clippy (Observability-Agent Client)
```

Comment:
nit: Can be more structured. I think it would be cleaner to do:
```
build module a
format code module a
[optional] test module a
clippy module a

[whitespace]

build module a
format code module a
[optional] test module a
clippy module a
```